eng/pipelines/ci.yml

```diff
@@ -94,27 +107,87 @@ extends:
                 displayName: 'Generation Task'
                 inputs:
                     BuildDropPath: $(System.DefaultWorkingDirectory)
-          
-          - job: publishAlpha
-            displayName: ESRP NPM Publish
-            dependsOn: buildJob
-            condition: eq(variables['Build.CronSchedule.DisplayName'], 'Nightly Alpha Release Build')     
+
+      - stage: releaseStage
+        dependsOn: buildStage
+        condition: or(eq(variables['Build.CronSchedule.DisplayName'], 'Nightly Alpha Release Build'), in(variables.releaseType, 'dev', 'next', 'latest'))
+        displayName: Release
+        jobs:
+          - job: releaseBuild
+            displayName: Build For Release
+            templateContext:
+              outputs:
+              - output: pipelineArtifact
+                path: $(System.DefaultWorkingDirectory)/
+            steps:
+              - task: Npm@1
+                displayName: 'Set postinstall scripts to true'
+                inputs:
+                  command: 'custom'
+                  customCommand: 'config set ignore-scripts false'
+              - task: NodeTool@0
+                displayName: 'Install Node LTS'
+                inputs:
+                  versionSpec: '22.x'
+              - task: PowerShell@2
+                displayName: 'Set alpha variables (dev tag)'
+                condition: eq(variables['Build.CronSchedule.DisplayName'], 'Nightly Alpha Release Build') # Only from the nightly trigger
+                inputs:  
+                  targetType: 'inline'
+                  script: |
+                    Write-Host "##vso[task.setvariable variable=releaseType;]dev"
+              - task: Npm@1
+                displayName: Update Package Version for nightly release
+                condition: eq(variables.releaseType, 'dev') # Only set the alpha build tag on dev-tagged builds
+                inputs:
+                  command: 'custom'
+                  customCommand: 'version prerelease --git-tag-version false --preid alpha-$(Date:yyyyMMdd))'
+              - task: Npm@1
+                displayName: "Install Dependencies"
+                inputs:
+                  command: "ci"
+              - task: Npm@1
```

Comment:
Can we reuse output from the build step instead of building the app again?

Reply:
The package.json is updated for the build version, so no.